backend/workflow_manager/file_execution/models.py

```diff
@@ -11,6 +12,45 @@
 MIME_TYPE_LENGTH = 128
 
 
+class WorkflowFileExecutionManager(models.Manager):
+    def get_or_create_file_execution(
+        self,
+        workflow_execution: WorkflowExecution,
+        file_name: str,
+        file_size: int,
+        file_hash: str,
+        file_path: Optional[str] = None,
+        mime_type: Optional[str] = None,
+    ):
+        """
+        Retrieves or creates a new input file record for a workflow execution.
+
+        Args:
+        workflow_execution: The `WorkflowExecution` object associated with this file
+        file_name: The name of the input file
+        file_size: The size of the file in bytes
+        file_hash: The hash of the file content
+        file_path: (Optional) The full path of the input file
+        mime_type: (Optional) MIME type of the file
+
+        return:
+            The `WorkflowFileExecution` object
+        """
+        execution_file: WorkflowFileExecution
+        execution_file, is_created = self.get_or_create(
+            workflow_execution=workflow_execution,
+            file_hash=file_hash,
+            file_path=file_path,
+        )
+
+        if is_created:
+            execution_file.file_name = file_name
+            execution_file.file_size = file_size
+            execution_file.mime_type = mime_type
+            execution_file.save()
```

Comment:
NIT: @muhammad-ali-e is it not possible to pass these as params to `get_or_create` rather than updating it again after saving it?

Reply:
I think we can keep it like this for now. The` workflow_execution, file_hash, and file_path` fields have an `index` and unique constraint, which helps optimize queries. However, if we add fields like `file_name`, `file_size`, and `mime_type` to the `get_or_create` query, it could lead to slower queries, especially if those fields are not indexed, as the database would need to perform more extensive lookups.